librz/bin/p/bin_omf166.c

```diff
@@ -0,0 +1,428 @@
+	if (!ret) {
+		return NULL;
+	}
+
+	RzBinMap *map = NULL;
+	RzPVector *v = obj->pe_vec;
+	void **it;
+	rz_pvector_foreach (v, it) {
+		OMF_pes *pe = (OMF_pes *)*it;
+
+		if (!(map = RZ_NEW0(RzBinMap))) {
+			rz_pvector_free(ret);
+			return NULL;
+		}
+		map->paddr = pe->paddr;
+		map->vaddr = (pe->SegmentNumber8 << 16) + pe->offset;
+		map->psize = map->vsize = pe->size;
+		map->perm = get_perm_by_type(pe->data_type);
+		map->name = rz_str_dup(get_data_type(pe->data_type));
+		rz_pvector_push(ret, map);
+	}
+	return ret;
+}
+
+static RzPVector /*<RzBinSection *>*/ *sections(RzBinFile *bf) {
+	RzPVector *ret;
+
+	if (!bf || !bf->o || !bf->o->bin_obj) {
+		return NULL;
+	}
+
+	if (!(ret = rz_pvector_new((RzPVectorFree)rz_bin_section_free))) {
+		return NULL;
+	}
+	rz_bin_omf166_obj *obj = bf->o->bin_obj;
+
+	RzPVector *v = obj->sections_vec;
+	void **it;
+	rz_pvector_foreach (v, it) {
+		OMF_sections *section = (OMF_sections *)*it;
+
+		RzBinSection *new = NULL;
+		if (!(new = RZ_NEW0(RzBinSection))) {
+			rz_pvector_free(ret);
+			return NULL;
+		}
+
+		const char *name = NULL;
+		OMF_lnames *lname = (OMF_lnames *)rz_pvector_at(obj->lnames_vec, section->index);
+		name = RZ_STR_ISNOTEMPTY(lname->name) ? lname->name : "NONE";
+		new->name = rz_str_dup(name);
+		new->size = 64 * 1024;
+		new->vsize = section->Seclen;
+		new->vaddr = (section->SegmentNumber8 << 16) + section->offset;
+		new->has_strings = false;
+		new->is_data = (section->Type == 1);
+		new->is_segment = 0;
+		new->perm = get_perm_by_type(section->Type);
+		rz_pvector_push(ret, new);
+	}
+	return ret;
+}
+
+static int offset_cmp(const void *a, const void *b, void *user) {
+	const OMF_symbol *sa = a;
+	const OMF_symbol *sb = b;
+	// first, sort by addr
+	if (sa->offset < sb->offset) {
+		return -1;
+	}
+	if (sa->offset > sb->offset) {
+		return 1;
+	}
+	return 0;
+}
+
+static RzPVector /*<RzBinSymbol *>*/ *symbols(RzBinFile *bf) {
+	RzPVector *ret;
+	RzBinSymbol *sym;
+
+	if (!bf || !bf->o || !bf->o->bin_obj) {
+		return NULL;
+	}
+
+	rz_bin_omf166_obj *obj = bf->o->bin_obj;
+
+	if (rz_pvector_len(obj->symbols_vec) < 1) {
+		return NULL;
+	}
+
+	if (!(ret = rz_pvector_new((RzPVectorFree)rz_bin_symbol_free))) {
+		return NULL;
+	}
+
+	RzPVector *v = obj->symbols_vec;
+	rz_pvector_sort(v, offset_cmp, NULL);
+	void **it;
+	rz_pvector_foreach (v, it) {
+		OMF_symbol *p = (OMF_symbol *)*it;
+		if (!p->is_data) {
+			if (!(sym = RZ_NEW0(RzBinSymbol))) {
+				return ret;
+			}
+			// sym->name = (p->ti == 0x4B) ? rz_str_newf("label.%s", p->name2) : rz_str_dup(p->name2); ///< ?
+			sym->name = rz_str_dup(p->name2); // ?
+			sym->forwarder = "NONE"; // ?
+
+			sym->paddr = p->offset; // ?
+			sym->vaddr = p->base + p->offset;
+			// sym->ordinal = ct_sym;
+			sym->size = p->size;
+			sym->is_imported = (p->ti == 0x004e);
+
+			switch (p->rec_type) {
+			case OMF166_GLBDEF:
+				sym->bind = RZ_BIN_BIND_GLOBAL_STR;
+				sym->type = RZ_BIN_TYPE_FUNC_STR;
+				break;
+			case OMF166_LOCSYM:
```

Comment:
i'm sure there is a bind for local symbol

Reply:
Fixed